crypto/wolfssl/Kconfig

```diff
@@ -0,0 +1,108 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+menuconfig CRYPTO_WOLFSSL
+	bool "wolfSSL SSL/TLS Cryptography Library"
+	default n
+	---help---
+		Enable support for wolfSSL
```

Comment:
I think wolfSSL should depends on CONFIG_ALLOW_GPL_COMPONENTS because is it not Apache license

Reply:
@jpbland1 how about this comment?

Reply:
@xiaoxiang781216 is that not the right way to do it?

Reply:
Yes, but GPL is an invasive license, we need a way to highlight it to the user.

Reply:
What do I need to do then? It already won't show up unless you enable GPL in the license select, would you like me to add something to the option text to make it obvious?

Reply:
> Yes, but GPL is an invasive license, we need a way to highlight it to the user. Anyway, I am fine without ALLOW_GPL_COMPONENTS. What's your opinion @acassis and @hartmannathan ?

Just make it depends on ALLOW_GPL_COMPONENTS. Users have to enable that explicitly because it is False by default. If they enable it, then they accept the GPL license will apply to their binaries.

Reply:
@hartmannathan I'm confused and maybe don't know enough about the build system, as it is currently the Kconfig option is wrapped with `if ALLOW_GPL_COMPONENTS` do I need to add something else to exclude it?

Reply:
> @hartmannathan I'm confused and maybe don't know enough about the build system, as it is currently the Kconfig option is wrapped with `if ALLOW_GPL_COMPONENTS` do I need to add something else to exclude it?

@jpbland1 Apologies for my confusing reply earlier. I was replying to @xiaoxiang781216, but on second reading, it looks like I misunderstood the comment.

I think this PR is fine the way it is because it does show as "Depends on: ALLOW_GPL_COMPONENTS" in kconfig-frontends.

I tested as follows:

```
$ cd ../apps/
$ wget -q https://github.com/apache/nuttx-apps/pull/1557.patch
$ git apply 1557.patch
$ cd ../nuttx/
$ tools/configure.sh tm4c129e-launchpad:nsh
$ make menuconfig
```

If I do not select ALLOW_GPL_COMPONENTS then CRYPTO_WOLFSSL is invisible in the menu:

```
    [ ] LibTomCrypt CrypographyLibrary  ----
    [ ] Mbed TLS Cryptography Library  ----
        Crypto: TinyCrypt cryptography library  --->
```

I can search for CRYPTO_WOLFSSL and it shows the GPL dependency: 

```
Symbol: CRYPTO_WOLFSSL [=n]
Type  : boolean
Prompt: wolfSSL SSL/TLS Cryptography Library
  Location:
    -> Application Configuration
(1)   -> Cryptography Library Support
  Defined at /home/x/NuttX/apps/crypto/wolfssl/Kconfig:7
  Depends on: ALLOW_GPL_COMPONENTS [=n]
```

If I select ALLOW_GPL_COMPONENTS then CRYPTO_WOLFSSL appears in the menu:

```
    [ ] LibTomCrypt CrypographyLibrary  ----
    [ ] Mbed TLS Cryptography Library  ----
        Crypto: TinyCrypt cryptography library  --->
    [ ] wolfSSL SSL/TLS Cryptography Library (NEW)  ----
```

The help text for this item shows the dependency.

I think that's fine.

If we want to have "depends on ALLOW_GPL_COMPONENTS" actually written in the stanza of menuconfig CRYPTO_WOLFSSL, it can be done with a simple refactoring: just remove the outermost "if ALLOW_GPL_COMPONENTS" ... "endif # ALLOW_GPL_COMPONENTS" and add "depends on ALLOW_GPL_COMPONENTS" under menuconfig CRYPTO_WOLFSSL. It will have the same effect as now. The only difference is that perhaps some people prefer that as more self-documenting? More future-proof against this Kconfig file being edited in the future? I don't have a strong opinion either way. I'm perfectly OK with either way.

Reply:
@jpbland1 sorry, I don't see you change at the begin, the patch looks fine now, let's merge it, thank you.